src/js/other-websites/inpage_menu.js

```diff
@@ -0,0 +1,249 @@
+  });
+
+  // Create "Get unlimited aliases" link
+  getUnlimitedAliasesBtn.href = `${relaySiteOrigin}/premium?utm_source=fx-relay-addon&utm_medium=input-menu&utm_content=get-premium-link`;
+
+  // Restrict tabbing to relay menu elements
+  restrictOrRestorePageTabbing(-1);
+
+  if (!premium) {
+    if (maxNumAliasesReached) {
+      generateAliasBtn.remove();
+      sendInPageEvent("viewed-menu", "input-menu-max-aliases-message");
+      remainingAliasesSpan.textContent = browser.i18n.getMessage(
+        "pageFillRelayAddressLimit",
+        [numAliasesRemaining, maxNumAliases]
+      );
+    }
+  } else {
+    getUnlimitedAliasesBtn.remove();
+  }
+
+  //Check if premium features are available
+  const premiumCountryAvailability = (
+    await browser.storage.local.get("premiumCountries")
+  )?.premiumCountries;
+
+  if (
+    premiumCountryAvailability?.premium_available_in_country !== true ||
+    !maxNumAliasesReached
+  ) {
+    getUnlimitedAliasesBtn.remove();
+  }
+
+  // Handle "Generate New Alias" clicks
+  generateAliasBtn.addEventListener("click", async (generateClickEvt) => {
+    sendInPageEvent("click", "input-menu-generate-alias");
+    preventDefaultBehavior(generateClickEvt);
+
+    // Request the active tab from the background script and parse the `document.location.hostname`
+    const currentPageHostName = await browser.runtime.sendMessage({method: "getCurrentPageHostname"});
+
+    // Attempt to create a new alias
+    const newRelayAddressResponse = await browser.runtime.sendMessage({
+      method: "makeRelayAddress",
+      description: currentPageHostName 
+    });
+
+    const loadingImagePath = browser.runtime.getURL("/images/loader.svg");
+    const loadingAnimationImage = document.querySelector(
+      ".fx-relay-alias-loading-image img"
+    );
+    loadingAnimationImage.src = loadingImagePath;
+
+    const relayInPageMenu = document.querySelector(".fx-relay-menu");
+
+    relayInPageMenu.classList.add("fx-relay-alias-loading");
+
+    // Catch edge cases where the "Generate New Alias" button is still enabled,
+    // but the user has already reached the max number of aliases.
+    if (newRelayAddressResponse.status === 402) {
+      relayInPageMenu.classList.remove("fx-relay-alias-loading");
+      // preserve menu height before removing child elements
+      // TODO: Add background/function to adjust height of iframe
+      relayInPageMenu.style.height = relayInPageMenu.clientHeight + "px";
+
+      [generateAliasBtn, remainingAliasesSpan].forEach((el) => {
+        el.remove();
+      });
+
+      const errorMessage = createElementWithClassList(
+        "p",
+        "fx-relay-error-message"
+      );
+      errorMessage.textContent = browser.i18n.getMessage(
+        "pageInputIconMaxAliasesError",
+        [relayAddresses.length]
+      );
+
+      relayInPageMenu.insertBefore(errorMessage, relayMenuDashboardLink);
+      return;
+    }
+
+    setTimeout(async () => {
```

Comment:
Do you happen to recall why this was wrapped in a `setTimeout`, and if that concern is still relevant?

Reply:
Not needed! Removed in 4344dc8e29ad93376f4af2c58eebb7a81182ed08